airbyte-lib/airbyte_lib/_executor.py

```diff
@@ -171,18 +180,23 @@ def ensure_installation(
         venv_path = Path(venv_name)
         if not venv_path.exists():
             if not self.install_if_missing:
-                raise Exception(
-                    f"Connector {self.metadata.name} is not available - "
-                    f"venv {venv_name} does not exist"
+                raise exc.AirbyteConnectorNotFoundError(
+                    message="Connector not available and venv does not exist.",
+                    context={
+                        "connector_name": self.metadata.name,
+                        "venv_name": venv_name,
+                    },
                 )
             self.install()
 
         connector_path = self._get_connector_path()
         if not connector_path.exists():
-            raise FileNotFoundError(
-                f"Could not find connector '{self.metadata.name}' in venv '{venv_name}' with "
-                f"connector path '{connector_path}'.",
-            )
+            raise exc.AirbyteConnectorNotFoundError(
+                connector_name=self.metadata.name,
+                context={
+                    "venv_name": venv_name,
```

Comment:
I think connector path is relevant debugging information here.

Reply:
I wrapped connector path and filenotfound in the "from" clause below. I think we should be good here.